### PR TITLE
Fix `ColorField` in RGB/RGBA format with `image_field`

### LINKED ISSUE #153: Crash when using `image_field` with RGB/RGBA formats
**Python version**
n/a

**Django version**
n/a

**Package version**
0.10.0

**Current behavior (bug description)**
The [recently added feature](https://github.com/fabiocaccamo/django-colorfield/pull/151) for RGB & RGBA doesn't work with the `image_field` feature. Issue noted https://github.com/fabiocaccamo/django-colorfield/issues/151#issuecomment-1706132604

> I realized just now that RGB and RGBA formats raise an error when the `image_field` option is used: https://github.com/fabiocaccamo/django-colorfield#image_field
> 
> This happens because the color returned by `get_image_background_color` (https://github.com/fabiocaccamo/django-colorfield/blob/main/colorfield/utils.py#L4) is always in HEX / HEXA format.

**Expected behavior**
RGB/RGBA format should work with `image_field`.

## Upvote & Fund

- We're using [Polar.sh](https://polar.sh/fabiocaccamo) so you can upvote and help fund this issue.
- We receive the funding once the issue is completed & confirmed by you.
- Thank you in advance for helping prioritize & fund our backlog.

<a href="https://polar.sh/fabiocaccamo/django-colorfield/issues/153">
<picture>
  <source media="(prefers-color-scheme: dark)" srcset="https://polar.sh/api/github/fabiocaccamo/django-colorfield/issues/153/pledge.svg?darkmode=1">
  Fund with Polar
</picture>
</a>

### PR DESCRIPTION
**Pull request by browniebroke**
---
name: Pull request
about: Submit a pull request for this project
assignees: fabiocaccamo

---

**Describe your changes**
Fix error when using image_field with RGB and RGBA formats

**Related issue**
Fix #153

**Checklist before requesting a review**
- [x] I have performed a self-review of my code.
- [x] I have added tests for the proposed changes.
- [x] I have run the tests and there are not errors.

### CODE REVIEW

File: colorfield/utils.py [hunk 1]
```diff
@@ -1,20 +1,33 @@
...
+    if img_format in {"hex", "hexa"}:
+        color_format = "#" + "%02x" * len(pixel_color)
+        color = color_format % pixel_color
+        color = color.upper()
+    elif img_format in {"rgb", "rgba"}:
+        if has_alpha:
+            # Normalize alpha channel to be between 0 and 1
+            pixel_color = (
+                *pixel_color[:3],
+                pixel_color[3] / 255,
```
Comment (fabiocaccamo) [inline colorfield/utils.py:17]: Should round alpha channel decimal places: `round(pixel_color[3] / 255, 2)`

**Review by fabiocaccamo — CHANGES_REQUESTED** [review]

File: tests/test_fields.py [hunk 2]
```diff
@@ -171,13 +172,28 @@ def test_model_with_image_and_format(self):
     def test_model_rgb_formats(self):
         obj = ColorFieldRGBFormat(
             color_rgb="rgb(123, 123, 123)",
-            color_rgba="rgba(5, 10, 128, 0.5)",
+            color_rgba="rgba(128, 199, 255, 0.55)",
```
Comment (browniebroke) [inline tests/test_fields.py:175]: Not sure why this passes, I would have expected that we need to increase the default `max_length`, as it's 25 chars long:

https://github.com/fabiocaccamo/django-colorfield/blob/a019212bcded27949b64e4fe280c3d4cf97dd25c/colorfield/fields.py#L44

Comment (browniebroke) [reply tests/test_fields.py:175]: Right... SQLite does not enforce the length of a VARCHAR

Comment (fabiocaccamo) [reply tests/test_fields.py:175]: At this point the field `max_length` value must be updated to 25.

**Review by fabiocaccamo — CHANGES_REQUESTED** [review]

**Review by fabiocaccamo — APPROVED** [review]
Thank you again also for this PR!

**Pull request merged**